VirtualList/VirtualListBase.js

```diff
@@ -27,8 +27,12 @@ const
 	isUp = is('up'),
 	JS = 'JS',
 	Native = 'Native',
-	// using 'bitwise or' for string > number conversion based on performance: https://jsperf.com/convert-string-to-number-techniques/7
-	getNumberValue = (index) => index | 0,
+	getNumberValue = (index) => {
```

Comment:
Note: this function is re-implemented to cover the case of _undefined_ string and to return _-1_ for invalid inputs. Still much faster than `parseInt`.